### PR TITLE
search functionality for map fragment

### PR DESCRIPTION
**Pull request by aurangzaibumer**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes https://github.com/onaio/fhir-resources/issues/3255

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by aurangzaibumer** [comment]
@ellykits this is currently in WIP state/Not finalized.

**Comment by aurangzaibumer** [comment]
Updating few things before marking ready for review.

**Comment by aurangzaibumer** [comment]
@ellykits I have separated the shared search functionality part from this PR, this PR only contains search functionality done on the map fragment.

**Comment by ellykits** [comment]
Can you reference the issue on this PR @aurangzaibumer?

File: android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/TopScreenSectionConfig.kt [hunk 1]
```diff
@@ -25,4 +25,5 @@ data class TopScreenSectionConfig(
   val searchBar: RegisterContentConfig?,
   val title: String? = null,
   val menuIcons: List<ImageProperties>? = null,
+  val shouldShowSearchButton: Boolean = false,
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/TopScreenSectionConfig.kt:28]:
```suggestion
  val showSearchButton: Boolean = false,
```

Better check if the `searchBar` configuration is `null`. If null do not display the search bar. The new property won't be necessary.

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/TopScreenSectionConfig.kt (hunk 1 above)
Comment (aurangzaibumer) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/TopScreenSectionConfig.kt:28]: This is for the extra button on searchBar like on Multi select widget which triggers the sync when clicked.

**Comment by aurangzaibumer** [comment]
> Can you reference the issue on this PR @aurangzaibumer?

done, sir.

**Comment by aurangzaibumer** [comment]
Updating tests...

**Comment by aurangzaibumer** [comment]
Closing this PR as @ellykits worked on a new implementation which will be replaced by this.

**Pull request closed without merging**